command-line-interface.md

```diff
@@ -0,0 +1,150 @@
+# Operations
+
+A conformant [runtime][] MUST provide an executable (called `funC` in the following examples).
+That executable MUST support commands with the following template:
+
+```sh
+$ funC [global-options] <COMMAND> [command-specific-options] <command-specific-arguments>
+```
+
+## Global options
+
+None are required, but the runtime MAY support options that start with at least one hyphen.
+Global options MAY take positional arguments (e.g. `--log-level debug`).
+Command names MUST NOT start with hyphens.
+The option parsing MUST be such that `funC <COMMAND>` is unambiguously an invocation of `<COMMAND>` (even for commands not specified in this document).
+If the runtime is invoked with an unrecognized command, it MUST exit with a nonzero exit code and MAY log a warning to stderr.
+
+## Character encodings
+
+This API specification does not cover character encodings, but runtimes SHOULD conform to their native operating system.
+For example, POSIX systems define [`LANG` and related environment variables][posix-lang] for [declaring][posix-locale-encoding] [locale-specific character encodings][posix-encoding], so a runtime in an `en_US.UTF-8` locale SHOULD write its [version](#version) to stdout in [UTF-8][].
+
+## Commands
+
+### version
```

Comment:
On Fri, Jul 08, 2016 at 09:12:46PM -0700, Aleksa Sarai wrote:
> > +### version
>
> For what it's worth, `--version` is the standard way GNU/Linux (or
> even UNIX commands in general) operate.

I think that's because most GNU/Linux/Unix commands are usually a
single command, not an umbrella with a number of subcommands.
Commands that are setup that way tend to support these sorts of things
using unhyphenated forms, although there's not a lot of consistency
:p.

* git(1) supports both --version and ‘version’, --help and ‘help’.
* btrfs(8) supports both --version and ‘version’, --help and ‘help’.
* ip(1) supports -V / -Version and ‘help’.
* npm supports ‘version’ and ‘help’.
* pip supports ‘--version’, --help and ‘help’.

I'm not sure any of those commands are aiming for spec-level
documentation.  For example, git(1) has:

  git [--version] … <command> [<args>]

even though you don't need a <command> when you set --version.

It seems like an easy thing to wrap either way (and if I was writing a
runtime I'd support both), but with a ‘version’ command we don't need
the “but you don't have to specify a command if you use one of these
command-like global options” caveat in this document.